src/plugins/home/public/application/components/homepage/homepage.tsx

```diff
@@ -0,0 +1,275 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useState, useEffect } from 'react';
+import { useMount } from 'react-use';
+import { Subscription } from 'rxjs';
+import { i18n } from '@osd/i18n';
+import { FormattedMessage } from '@osd/i18n/react';
+import {
+  EuiPageTemplate,
+  EuiButtonEmpty,
+  EuiHorizontalRule,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiLoadingSpinner,
+  EuiIcon,
+} from '@elastic/eui';
+import {
+  HeroSection as HeroSectionType,
+  Section as SectionType,
+} from '../../../services/section_type/section_type';
+import { getServices } from '../../opensearch_dashboards_services';
+import { HeroSection } from './hero_section';
+import { Section } from './section';
+import { Footer } from './footer';
+import { Welcome } from '../welcome';
+
+const KEY_ENABLE_WELCOME = 'home:welcome:show';
+
+const useHomepage = () => {
+  const { sectionTypes } = getServices();
+  const [heroes, setHeroes] = useState<HeroSectionType[]>();
+  const [sections, setSections] = useState<SectionType[]>();
+  const [error, setError] = useState<unknown>();
+
+  useEffect(() => {
+    const homepage = sectionTypes.getHomepage();
+
+    const subscriptions = new Subscription();
+    subscriptions.add(homepage.heroes$.subscribe(setHeroes));
+    subscriptions.add(homepage.sections$.subscribe(setSections));
+    subscriptions.add(homepage.error$.subscribe(setError));
+
+    return () => {
+      subscriptions.unsubscribe();
+      homepage.cleanup();
+    };
+  }, [sectionTypes]);
+
+  const isLoading = !heroes && !sections;
+
+  return { heroes, sections, error, isLoading };
+};
+
+const useShowWelcome = () => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [isNewInstance, setIsNewInstance] = useState(false);
+  const { homeConfig, savedObjectsClient } = getServices();
+
+  const isWelcomeEnabled = !(
+    homeConfig.disableWelcomeScreen || localStorage.getItem(KEY_ENABLE_WELCOME) === 'false'
+  );
+
+  useEffect(() => {
+    if (!isWelcomeEnabled) {
+      setIsLoading(false);
+      return;
+    }
+
+    const timeout = setTimeout(() => {
+      setIsLoading(false);
+    }, 500);
```

Comment:
Is 500ms a good enough timeout?

Reply:
This is copied over from the previous implementation:

https://github.com/opensearch-project/OpenSearch-Dashboards/blob/b5d39b6e651ecf6c32fc87d6f8f893a86d48e3f2/src/plugins/home/public/application/components/home.js#L96-L102

Reply:
Do you think we even need this? What I would hate is for a spinner to flash on the screen and vanish. 500ms is too short for anyone to see anything or to be meaningful.

Reply:
There are a few paths we could follow:
 - The user has already disable the welcome screen. This could occur if the config toggle is set or the user previously dismissed it. This is executed immediately, and involves zero  re-renders. Zero flashing will occur.
 - The instance is new and the welcome screen isn't disabled. To figure out if the instance is new, we query the saved objects client for any index patterns. This involves a short loading period while we wait for the saved objects client to provide us with a response. While we wait, the page will be blank.
 - There is an error getting a response from the saved objects client or it takes too long. This is the case that the timeout accounts for. If we don't get a success response in 500 ms, we load the page as usual.

So the outcome of this logic is that we either
 - show the page immediately or
 - load until we know whether or not the welcome screen should be shown

I don't know about the reliability of `savedObjectsClient.find`, but that's basically what we're questioning here.

Reply:
Correct me if I am wrong: if the call to savedObjects takes longer than 500ms (from the browser) we load the page but show re-render when the result comes through? As in, this 500ms is just to make sure we don't re-render immediately on page load?

Reply:
You helped me catch a bug. You're correct, that was the functionality. I just updated it so that no matter what, if the response doesn't come in 500ms, the welcome screen is not shown. The 500ms is basically just to make sure we don't block the page load if `savedObjectsClient.find` doesn't return in time.